ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -1,257 +1,216 @@
 
-    Returns
-    -------
-        Callable[[int],Callable]: Callable Encoder for scalar types that takes
-        in field_index and returns the Low-level Encode Callable.
 
+def _scalar_encoder(encoder: EncoderConstructor) -> PartialEncoderConstructor:
+    """Constructs a scalar encoder constructor.
+
+    Takes a field index and returns an Encoder.
+
+    This class returns the Encoder with is_repeated set to False
+    and is_packed set to False.
     """
 
-    def scalar_encoder(field_index):
+    def scalar_encoder(field_index: int) -> Encoder:
         is_repeated = False
         is_packed = False
         return encoder(field_index, is_repeated, is_packed)
 
     return scalar_encoder
 
 
-def _vector_encoder(encoder, is_packed=True) -> Callable[[int], Callable]:
-    """Abstract Specific Encoder(Callable) as Vector Encoder Callable that takes in field index.
-
-    Args
-    ----
-       encoder (Callable[[int, bool, bool], Callable]): Specific encoder(Callable) that takes in
-       field_index, is_repeated, is_packed and returns the Low-level Encode Callable.
+def _vector_encoder(
+    encoder: EncoderConstructor, is_packed: bool = True
+) -> PartialEncoderConstructor:
+    """Constructs a vector (array) encoder constructor.
 
-       is_packed (bool, optional): Represents if the encoder supports packed type. Defaults to True.
-
-    Returns
-    -------
-        Callable[[int],Callable]: Callable Encoder for 1D Array types that takes in
-        field_index and returns the Low-level Encode Callable.
+    Takes a field index and returns an Encoder.
 
+    This class returns the Encoder with is_repeated set to True
+    and is_packed defaulting to True.
     """
 
-    def vector_encoder(field_index):
+    def vector_encoder(field_index: int) -> Encoder:
         is_repeated = True
         return encoder(field_index, is_repeated, is_packed)
 
     return vector_encoder
 
 
-def _scalar_decoder(decoder) -> Callable[[int, str], Callable]:
-    """Abstract Specific Decoder(Callable) as Scalar Decoder Callable that takes in field index,key.
-
-    Args
-    ----
-        decoder (Callable[[int, bool, bool], Callable]): Specific decoder(Callable) that takes in
-        field_index, is_repeated, is_packed,  key, new_default and
-        returns the Low-level Decode Callable.
+def _scalar_decoder(decoder: DecoderConstructor) -> PartialDecoderConstructor:
+    """Constructs a scalar decoder constructor.
 
-    Returns
-    -------
-        Callable[[int,str],Callable]: Callable Decoder for scalar types that takes in
-        field_index, key and returns the Low-level Decode Callable.
+    Takes a field index and a key and returns a Decoder.
 
+    This class returns the Decoder with is_repeated set to False
+    and is_packed set to False.
     """
 
-    def scalar_decoder(field_index, key):
+    def _unsupported_new_default(message: Message) -> Message:
```

Comment:
Match the other one.
```suggestion
    def _unsupported_new_default(message: Optional[Message] = None) -> Any:
```

Reply:
Return value is Any